Formula/docker-credential-magic.rb

```diff
@@ -0,0 +1,58 @@
+class DockerCredentialMagic < Formula
+  desc "Magic shim for Docker credential helpers"
+  homepage "https://github.com/docker-credential-magic/docker-credential-magic"
+  url "https://github.com/docker-credential-magic/docker-credential-magic/archive/v0.8.0.tar.gz"
+  sha256 "939e12f0d6514c62877ca20cfac78f4556de1353ca4defa34b67df660b5bc1d4"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  on_macos do
+    if Hardware::CPU.intel?
+      url "https://github.com/docker-credential-magic/docker-credential-magic/archive/v0.8.0.tar.gz"
+      sha256 "939e12f0d6514c62877ca20cfac78f4556de1353ca4defa34b67df660b5bc1d4"
+
+      def install
+        system "go", "mod", "verify"
+        system "make", "clean", "vendor", "fetch-helpers", "copy-mappings", "build-magic-embedded"
+        system "make", "build-magic"
+        system "make", "build-magician"
+      end
+    end
+  end
+
+  on_linux do
+    if Hardware::CPU.arm? && !Hardware::CPU.is_64_bit?
+      url "https://github.com/docker-credential-magic/docker-credential-magic/releases/download/v0.8.0/docker-credential-magic_Linux_armv6.tar.gz"
+      sha256 "fa6deaf643623a163fafdc5e27717207fc2c9fc13da03e59a001194c9f951f41"
+
+      def install
+        bin.install "docker-credential-magic"
+        bin.install "docker-credential-magician"
+      end
+    end
+    if Hardware::CPU.arm? && Hardware::CPU.is_64_bit?
+      url "https://github.com/docker-credential-magic/docker-credential-magic/releases/download/v0.8.0/docker-credential-magic_Linux_arm64.tar.gz"
+      sha256 "4ec0f0630c1d91a9025f71f34dbbf678bc2553fcae956d6927d0431d70e0bf68"
+
+      def install
+        bin.install "docker-credential-magic"
+        bin.install "docker-credential-magician"
+      end
+    end
+    if Hardware::CPU.intel?
+      url "https://github.com/docker-credential-magic/docker-credential-magic/releases/download/v0.8.0/docker-credential-magic_Linux_x86_64.tar.gz"
+      sha256 "afbb0aacea8b5cbdf9f99cd3ec5a9f649ee421866ec50894bc83ea30e0b812f9"
+
+      def install
+        bin.install "docker-credential-magic"
+        bin.install "docker-credential-magician"
+      end
+    end
+  end
```

Comment:
```suggestion
```
We need to build this from source everywhere.

Reply:
thank you so much for the tips 🙋🏻‍♂️✌️